astropy/modeling/functional_models.py

```diff
@@ -400,13 +400,15 @@ def evaluate(x, factor):
         return factor * x
 
 
-class Redshift(Fittable1DModel):
+class RedshiftWavelength(Fittable1DModel):
     """
     One dimensional redshift model.
 
+    Redshifts a wavelength array from rest (z=0) to the input redshift.
+
     Parameters
     ----------
-    z : float or a list of floats
+    z : int or float
```

Comment:
Really any parameter for any model can take any scalar or array of numeric values (though I don't think complex is supported).  So even just saying "list of floats" is misleading because anything that can be converted to an `ndarray` is fine.  As @nden wrote saying `int` is certainly possible but ultimately not much point.  

Honestly it's been on my TODO list (albeit low priority) to go through and normalize all the model docstrings so that parameter types are documented consistently.